pkg/services/ngalert/notifier/channels/alertmanager.go

```diff
@@ -25,24 +25,20 @@ func NewAlertmanagerNotifier(model *NotificationChannelConfig, _ *template.Templ
 	if model.SecureSettings == nil {
 		return nil, receiverInitError{Cfg: *model, Reason: "no secure settings supplied"}
 	}
-	urlStr := model.Settings.Get("url").MustString()
-	if urlStr == "" {
-		return nil, receiverInitError{Reason: "could not find url property in settings", Cfg: *model}
+	if valid, err := ValidateContactPointReceiverWithSecure(model.Type, model.Settings, model.SecureSettings, fn); err != nil || !valid {
+		return nil, receiverInitError{Cfg: *model, Reason: err.Error()}
```

Comment:
I think we should separate two situations: error and invalid or change the signature of the validate method to return just error